### PR TITLE
auth

### PR DESCRIPTION
**Pull request by JackScher**
register & reset password

### CODE REVIEW

File: app/templates/auth/forgot.html [hunk 1]
```diff
@@ -1,45 +1,111 @@
...
-          method="post"
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>{{ config.APP_NAME }}</title>
+    <!-- meta -->
+    <meta name="description" content="Simple2B Flask App" />
+    <meta name="author" content="Simple2B" />
+    <meta name="viewport" content="width=device-width,initial-scale=1" />
```
Comment (kipod) [inline app/templates/auth/forgot.html]: WRONG WAY.
In the project, we have a base template: `base.html`, all other templates need extent it...

File: app/templates/auth/forgot.html [hunk 2]
```diff
@@ -1,45 +1,111 @@
...
-            </button>
+              <path
+                fill-rule="evenodd"
+                d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
+                clip-rule="evenodd"
+              ></path>
+            </svg>
+          </button>
+        </div>
+        <!--  -->
```
Comment (kipod) [inline app/templates/auth/forgot.html]: what do this line?

File: app/templates/auth/forgot.html [hunk 3]
```diff
@@ -1,45 +1,111 @@
...
+              Type your email, and we send your password
+            </h2>
+            <form class="mt-8 space-y-6 from"
+                  role="form"
+                  action="{{ url_for('auth.forgot_pass') }}"
+                  method="post">
+                {{ form.hidden_tag() }}
+
+        <div>
+              {{form.email.label(class='block mb-2 text-sm font-medium text-gray-900 dark:text-white')}} {{form.email(class='bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500')}}{% for error
```
Comment (kipod) [inline app/templates/auth/forgot.html]: use `<!-- prettier-ignore -->`

File: app/templates/auth/login.html [hunk 4]
```diff
@@ -11,6 +11,66 @@
...
     <link href="{{ url_for('static', filename='css/main.css') }}" rel="stylesheet"/>
       <body class="bg-gray-50 dark:bg-gray-900">
+        {% with messages = get_flashed_messages(with_categories=true) %}
+        {% if messages %}
+        {% for category, message in messages %}
+        <!-- Flash message markup -->
+        <div
+          id="toast-{{category}}"
+          class="absolute top-10 left-10 z-50 flex items-center w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
+          role="alert"
```
Comment (kipod) [inline app/templates/auth/login.html]: need extend base template...

File: app/templates/auth/register.html [hunk 5]
```diff
@@ -11,6 +11,66 @@
...
+            class="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-green-500 bg-green-100 rounded-lg dark:bg-green-800 dark:text-green-200"
+          >
+            <svg
+              aria-hidden="true"
+              class="w-5 h-5"
+              fill="currentColor"
+              viewBox="0 0 20 20"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path
```
Comment (kipod) [inline app/templates/auth/register.html]: need extend the base template

File: app/templates/auth/reset_password.html [hunk 6]
```diff
@@ -1,57 +1,117 @@
...
+    <meta charset="utf-8" />
+    <title>{{ config.APP_NAME }}</title>
+    <!-- meta -->
+    <meta name="description" content="Simple2B Flask App" />
+    <meta name="author" content="Simple2B" />
+    <meta name="viewport" content="width=device-width,initial-scale=1" />
+    <!-- styles -->
+    <!-- prettier-ignore -->
+    <link href="{{ url_for('static', filename='css/main.css') }}" rel="stylesheet"/>
+      <body class="bg-gray-50 dark:bg-gray-900">
```
Comment (kipod) [inline app/templates/auth/reset_password.html]: return code this `{% extends 'base.html' %} {% block content %}`

File: tests/test_auth.py [hunk 7]
```diff
@@ -41,7 +41,6 @@ def test_register(client):
             in response.data
         )
 
-        assert "bg-green-500" in response.data.decode()
```
Comment (kipod) [inline tests/test_auth.py:44]: 👍 super fix!

**Review by kipod — CHANGES_REQUESTED** [review]
Looks like joke

File: app/templates/auth/forgot.html [hunk 8]
```diff
@@ -1,111 +1,65 @@
...
-        </div>
-        <!--  -->
+    <button type="button" class="ml-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700" data-dismiss-target="#toast-{{category}}" aria-label="Close">
+      <span class="sr-only">Close</span>
+      <svg aria-hidden="true" class="w-5 h-5" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path>
+      </svg>
+    </button>
+  </div>
+  {% endfor %} {% endif %} {% endwith %}
```
Comment (StoliarskyiKostiantyn) [inline app/templates/auth/forgot.html]: We don't need those flash messages in this pages, we already have them in base, you need to remove them from here

File: app/templates/auth/register.html [hunk 9]
```diff
@@ -1,133 +1,111 @@
...
+      <span class="sr-only">Close</span>
+      <svg aria-hidden="true" class="w-5 h-5" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path>
+      </svg>
+    </button>
+  </div>
+  <!-- prettier-ignore -->
+  {% endfor %}
+  {% endif %}
+  {% endwith %}
```
Comment (StoliarskyiKostiantyn) [inline app/templates/auth/register.html]: same here

File: app/templates/auth/register.html [hunk 10]
```diff
@@ -1,133 +1,111 @@
...
+            </div>
+            <div>
+              {{form.password.label(class='block mb-2 text-sm font-medium
+              text-gray-900 dark:text-white')}}
+              {{form.password(class='bg-gray-50 border border-gray-300
+              text-gray-900 text-sm rounded-lg focus:ring-blue-500
+              focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700
+              dark:border-gray-600 dark:placeholder-gray-400 dark:text-white
+              dark:focus:ring-blue-500 dark:focus:border-blue-500')}}{% for
+              error in form.errors.password %}
```
Comment (StoliarskyiKostiantyn) [inline app/templates/auth/register.html:47]: Please pass more  to make code look more readalbe (look for example in base and login)

File: app/templates/auth/reset_password.html [hunk 11]
```diff
@@ -1,117 +1,85 @@
...
+  <!-- prettier-ignore -->
+  {% if messages %} 
+    {% for category, message in messages %}
+  <!-- Flash message markup -->
+  <!-- prettier-ignore -->
+  <div id="toast-{{category}}" class="absolute top-10 left-10 z-50 flex items-center w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800" role="alert">
+      <div class="inline-flex items-center justify-center flex-shrink-0 w-8 h-8 text-green-500 bg-green-100 rounded-lg dark:bg-green-800 dark:text-green-200">
+        <svg aria-hidden="true" class="w-5 h-5" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+          <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
+        </svg>
```
Comment (StoliarskyiKostiantyn) [inline app/templates/auth/reset_password.html]: same about flash messages

File: tests/test_auth.py [hunk 12]
```diff
@@ -41,6 +41,7 @@ def test_register(client):
             in response.data
         )
 
+        assert "toast" in response.data.decode()
```
Comment (StoliarskyiKostiantyn) [inline tests/test_auth.py:44]: i think we could add test for different toasts

**Review by StoliarskyiKostiantyn — CHANGES_REQUESTED** [review]

**Review by StoliarskyiKostiantyn — APPROVED** [review]

**Pull request merged**